sdk/identity/azure-identity/src/managed_identity_source.cpp

```diff
@@ -183,13 +186,19 @@ AppServiceManagedIdentitySource::AppServiceManagedIdentitySource(
 
     url.AppendQueryParameter("api-version", apiVersion);
 
-    // Only one of clientId or resourceId will be set to a non-empty value.
+    // Only one of clientId, objectId, or resourceId will be set to a non-empty value.
     // AppService uses mi_res_id, and not msi_res_id:
     // https://learn.microsoft.com/azure/app-service/overview-managed-identity?tabs=portal%2Chttp#rest-endpoint-reference
+    // Based on the App Service documentation, using principal_id for the query parameter name here
+    // instead of object_id (which is used as an alias).
     if (!clientId.empty())
     {
       url.AppendQueryParameter(clientIdHeaderName, clientId);
```

Comment:
It's not a big deal, but given that AppServiceV2017ManagedIdentitySource and AppServiceV2019ManagedIdentitySource are both derived from `AppServiceManagedIdentitySource`, I might have put `clientIdHeaderName`, `apiVersion`, and `secretHeaderName` as pure virtual methods on `AppServiceManagedIdentity` rather than requiring these as fixed constant parameters.

Water under the bridge. But something to consider when defining protocol version specializations of a class - IMHO it's better to isolate the protocol specific elements in the specializations rather than making those variations method parameters on the base class.

Reply:
Agreed! I think there are some improvements we can make on the class hierarchy.